src/pages/resultsView/download/DownloadUtils.ts

```diff
@@ -321,6 +322,36 @@ export function generateOtherMolecularProfileDownloadData(
         : [];
 }
 
+export function generateGenericAssayProfileData(
+    molecularProfileId: string[],
+    unfilteredCaseAggregatedData?: CaseAggregatedData<GenericAssayData>
+): { [key: string]: GenericAssayData[] } {
+    const sampleFilter = (data: GenericAssayData) => {
+        return molecularProfileId.includes(data.molecularProfileId);
+    };
+
+    return unfilteredCaseAggregatedData
+        ? generateGenericAssayDataByStableId(
+              unfilteredCaseAggregatedData,
+              sampleFilter
+          )
+        : {};
+}
+
+export function generateGenericAssayProfileDownloadData(
+    sampleAlterationDataByStableId: { [key: string]: GenericAssayData[] },
+    samples: Sample[] = [],
+    stableIds: string[] = []
+): string[][] {
+    return sampleAlterationDataByStableId
+        ? generateGenericAssayDownloadData(
```

Comment:
I think we can combine `generateGenericAssayProfileDownloadData` and `generateGenericAssayDownloadData`

Reply:
Sure!